### PR TITLE
PorousFlow JOSS changes

### LINKED ISSUE #15539: Some JOSS PorousFlow changes
## Reason

The suggestions below come from the JOSS editor ( https://github.com/openjournals/joss-reviews/issues/2176#issuecomment-653517195 ) .   They need to be done in porous_flow/paper.md and porous_flow/paper.bib

## Design

- [x] There is a period missing after the Lime reference (end of paragraph).
- [x] "simply solve simplistic" -> "only solve simplistic" or some other phrasing.
- [x] Could you make any comments in the comparison with other software, perhaps in terms of capability, performance, and easy of use/extension?
- [x] You're welcome to include a figure if you'd like.
- [x] Would it make sense to write down the basic equations that are solved? Perhaps one or two lines for each of THMC just to get the structural gist across, or a figure representing the role and coupling. (It's sort of abrupt to launch directly into prior applications).
- [x] Can you double-check capitalization in your references? Our toolchain uses pandoc-citeproc, which is based on apa.csl. Its semantics are closer to biblatex than bibtex, which is required to be precisely APA-conforming. Anyway, there are extra braces needed, e.g., `{I}celand` or `{Iceland}` are needed.

Also:

```
Reference check summary:

OK DOIs

- 10.1007/s12665-012-1546-x is OK
- 10.1007/s00366-006-0049-3 is OK
- 10.1080/22020586.2019.12073198 is OK

MISSING DOIs

- https://doi.org/10.1016/j.softx.2020.100430 may be missing for title: MOOSE: Enabling Massively Parallel Multiphysics Simulation
- https://doi.org/10.21203/rs.2.17644/v1 may be missing for title: Impact of Effective Normal Stress on Capillary Pressure in a Single Natural Fracture

INVALID DOIs

- None
```

### PR DESCRIPTION
**Pull request by WilkAndy**
Fixes #15539

- Regarding the equations: I've included some, although we wanted this paper to be readable by a more general audience, so have included the equations towards the end of the paper.

- Regarding comparisons with other codes: I've included another paragraph that discusses some of the benefits of using MOOSE+PorousFlow.  It's really hard to write this sort of thing without appearing to denigrate other very worthy codes!

- Done the DOIs mentioned above

### CODE REVIEW

**Comment by WilkAndy** [comment]
Hey @cpgr , could you please check this?  If you want to see what it looks like

https://whedon.theoj.org/
Paper repo: https://github.com/WilkAndy/moose
Branch : joss_pf_15539

Perhaps @jedbrown would also like to review the changes here?

a

**Comment by cpgr** [comment]
I just generated the paper and I'll have a good read this afternoon. I'll have to double check but I think I published that figure in another article. I remember making multiple versions, so will have a similar one to swap in if necessary.

**Comment by WilkAndy** [comment]
Thanks @cpgr

**Comment by WilkAndy** [comment]
> I just generated the paper and I'll have a good read this afternoon. I'll have to double check but I think I published that figure in another article. I remember making multiple versions, so will have a similar one to swap in if necessary.

By the way, we don't *NEED* to include a figure.  @jedbrown just suggested it as a possibility.

File: modules/porous_flow/paper.md [hunk 1]
```diff
@@ -23,28 +23,72 @@ date: 13 February 2020
...
 
-In many of these cases it is not enough to simply solve simplistic, traditional flow and transport physics.  Coupling with other physics is necessary: solid mechanical deformations and stresses can be important; geochemistry can alter the flow characteristics of the subsurface; high-precision equations of state are required; the evolution of fluid components (tracers, pollutants, reactants) is of interest; unsaturated physics is crucial.  The flow and transport often impacts the subsurface structure, and conversely, the effects of these structure-changes impact the flow and transport.
+In many of these cases it is not enough to only solve simplistic, traditional flow and transport physics.  Coupling with other physics is necessary: solid mechanical deformations and stresses can be important; geochemistry can alter the flow characteristics of the subsurface; high-precision equations of state are required; the evolution of fluid components (tracers, pollutants, reactants) is of interest; unsaturated physics is crucial.  The flow and transport often impacts the subsurface structure, and conversely, the effects of these structure-changes impact the flow and transport.
```
Comment (cpgr) [inline modules/porous_flow/paper.md:40]: Maybe remove the word simplistic here altogether?

File: modules/porous_flow/paper.md [hunk 2]
```diff
@@ -23,28 +23,72 @@ date: 13 February 2020
...
+Harnessing the power of MOOSE means that ``PorousFlow`` simulations efficiently use multiprocessors and automatically utilize threading/OpenMP, without developers or users having to worry about the software technicalities.  Mesh adaptivity with a variety of integration schemes and unstructured meshes may be used.  Various time-stepping schemes may be employed.  A variety of I/O formats are available, such as Exodus and VTK files as well as CSV plaintext.  The flexible plug-in architecture makes it easy for novice developers to solve new physics equations or couple together existing modules to solve complex problems.  Users may even utilising the "MultiApp" system that allows simulations to be embedded inside other simulations.  A GUI helps users create, run and analyse models.  An example ``PorousFlow`` result is shown in \autoref{fig:example}.
```
Comment (cpgr) [inline modules/porous_flow/paper.md:89]: `Users may even utilising the` -> `Users may utilise the` perhaps?

I guess no need to expand on I/O or gui as the readership is targeted towards software users.

File: modules/porous_flow/paper.md [hunk 3]
```diff
@@ -23,28 +23,72 @@ date: 13 February 2020
...
+
+A comprehensive listing of equations, including fluid equations of state, the evolution of porosity and permeability, relative-permeability functions, capillary relationships, chemical reactions, etc, may be found at https://mooseframework.org/modules/porous_flow/index.html.
+
+# Software details and comparisons
+
+``PorousFlow`` is built upon the open-source, massively parallel, fully implicit multiphysics simulation framework MOOSE (Multiphysics Object-Oriented Simulation Environment) [@permann2020moose].  MOOSE is an open-source library from Idaho National Laboratory that provides a high-level interface to the libMesh finite element library [@libmesh] and PETSc nonlinear solvers [@petsc].  MOOSE and ``PorousFlow`` follow [strict quality controls](https://mooseframework.org/sqa/index.html).
+
 Other open-source codes that can also perform tightly-coupled THMC simulations to some degree include OpenGeoSys [@opengeosys], PFLOTRAN [@pflotran] and work under the Open Porous Media initiative [@opm].
```
Comment (cpgr) [inline modules/porous_flow/paper.md:87]: I know what you meant about not underselling or denigrating the other capable codes here. Perhaps we could add something along the lines of `One aspect that makes PorousFlow different to these codes is the ease of coupling additional physics to the base flow and transport capabilities, and the relative speed and simplicity of developing new capability using the provided APIs` or something along those lines.

**Comment by cpgr** [comment]
> > I just generated the paper and I'll have a good read this afternoon. I'll have to double check but I think I published that figure in another article. I remember making multiple versions, so will have a similar one to swap in if necessary.
> 
> By the way, we don't _NEED_ to include a figure. @jedbrown just suggested it as a possibility.

It looks like I didn't publish that figure after all

File: modules/porous_flow/paper.md (hunk 1 above)
Comment (WilkAndy) [reply modules/porous_flow/paper.md:40]: Thanks, done

File: modules/porous_flow/paper.md (hunk 2 above)
Comment (WilkAndy) [reply modules/porous_flow/paper.md:89]: Thanks, yep, i agree

File: modules/porous_flow/paper.md (hunk 3 above)
Comment (WilkAndy) [reply modules/porous_flow/paper.md:87]: Thanks, i've added that and slightly modified the paragraph following it.

**Comment by WilkAndy** [comment]
> > > I just generated the paper and I'll have a good read this afternoon. I'll have to double check but I think I published that figure in another article. I remember making multiple versions, so will have a similar one to swap in if necessary.
> > 
> > 
> > By the way, we don't _NEED_ to include a figure. @jedbrown just suggested it as a possibility.
> 
> It looks like I didn't publish that figure after all

OK, that's surprising, since it's such a nice figure.  Thanks.

**Comment by cpgr** [comment]
Thanks for looking after this Andy. It's a nice little paper!

**Comment by WilkAndy** [comment]
> Thanks for looking after this Andy. It's a nice little paper!

:-)

File: modules/porous_flow/paper.bib [hunk 4]
```diff
@@ -78,33 +84,33 @@ @techreport{c25065
 }
 
 @inproceedings{green2018,
-title={Community Code for Simulating {CO}$_2$ Storage: Modelling Multiphase Flow with Coupled Geomechanics and Geochemistry Using the Open-Source Multiphysics Framework {MOOSE}},
+title={{Community Code for Simulating {CO}$_2$ Storage: Modelling Multiphase Flow with Coupled Geomechanics and Geochemistry Using the Open-Source Multiphysics Framework MOOSE}},
```
Comment (jedbrown) [inline modules/porous_flow/paper.bib:87]: APA style uses sentence case for titles so no need to make changes like this.

**Comment by WilkAndy** [comment]
Thanks @jedbrown , i think i've removed all superfluous `{{blah}}`

File: modules/porous_flow/paper.bib [hunk 5]
```diff
@@ -33,13 +33,19 @@ @misc{opm
...
 
 
-@misc{permann2019moose,
-    title={MOOSE: Enabling Massively Parallel Multiphysics Simulation},
-    author={Cody J. Permann and Derek R. Gaston and David Andrs and Robert W. Carlsen and Fande Kong and Alexander D. Lindsay and Jason M. Miller and John W. Peterson and Andrew E. Slaughter and Roy H. Stogner and Richard C. Martineau},
-    year={2019},
-    eprint={1911.04488},
-    archivePrefix={arXiv},
-    primaryClass={cs.MS}
+@article{permann2020moose,
```
Comment (aeslaughter) [inline modules/porous_flow/paper.bib:36]: There should be no need for this reference, it is already in framework/doc/content/bib/moose.bib, which the porous flow documentation can use. I will fix this up for you.

**Comment by aeslaughter** [comment]
I think I need to be educated, what are the paper.md/paper.bib files being used for here? I thought they were part of normal documentation, but they are outside of that.

File: modules/porous_flow/paper.bib (hunk 5 above)
Comment (aeslaughter) [reply modules/porous_flow/paper.bib:36]: I think you  can ignore this, you seem to have something else going on here outside of normal docs.

Comment (jedbrown) [reply modules/porous_flow/paper.bib:36]: It needs to be picked up by the JOSS generator, which uses `pandoc-citeproc`. One could reference a common bib file, but there's probably no particular need.

**Comment by WilkAndy** [comment]
> I think I need to be educated, what are the paper.md/paper.bib files being used for here? I thought they were part of normal documentation, but they are outside of that.

You are correct @aeslaughter .   paper.md and paper.bib and paper.png are part of the Journal of Open Source Software (JOSS) paper we are preparing.  So those three files are outside the normal documentation.  When our JOSS paper has been published they'll be renamed so that other modules can also create a paper.* in their directories and JOSS will pick them up (JOSS just searches the entire repo for paper.md and paper.bib and produces a PDF from them).   

Long, waffly explanation..... conclusion is that paper.bib should be as i wrote it (including the moose2020 paper).

**Review by aeslaughter — APPROVED** [review]

**Comment by WilkAndy** [comment]
Thanks @cpgr , @jedbrown and @aeslaughter for your help with this PR

a

**Pull request merged**